src/CSTools/Tools/parser.cs

```diff
@@ -1554,7 +1554,7 @@ SYMBOL Parse()
 			ParseStackEntry top = new ParseStackEntry(this,0,NextSym());
 			for (;;)
 			{
-				string cnm = top.m_value.yyname();
+				string cnm = top.m_value!.yyname();
```

Comment:
this could be null because `top` is reassigned in the call to `Pop(ref top ...` meaning `m_value` could be null here. I say we just explicitly throw a null error before this line if `top.m_value` is null as that's what it would have done before.

Reply:
But it returns after that call. It shouldn't ever do another iteration in this loop if it gets the successful parse. Right?

Reply:
oh you're right, man this code is just confusing. I'd still be inclined to throw as that's always going to be a better error then what gets thrown automatically, and then it's obvious to anyone looking at the code what will happen if it's null here.

Reply:
Sounds good